app/assets/javascripts/components/revisions/diff_viewer.jsx

```diff
@@ -156,11 +164,43 @@ const DiffViewer = React.createClass({
     const wikiDiffUrl = this.webDiffUrl();
 
     let diffComment;
-    // Only show the edit summary for a single revision; for cumulative diffs,
-    // the edit summary is for the *parent* of the first revision, so it's
-    // irrelevant and confusing to show that.
+    let revisionDateTime;
+    let firstRevTime;
+    let lastRevTime;
+    let days;
+    let hours;
+    let minutes;
+    let timeSpan = '';
+
+    // Edit summary for a single revision:
+    //  > Edit date and number of characters added
+    // Edit summary for range of revisions:
+    //  > Span of time for edits to article (from first applicable rev to last)
     if (!this.props.first_revision) {
-      diffComment = <p className="diff-comment">{this.state.comment}</p>;
+      revisionDateTime = moment(this.props.revision.date).format('YYYY/MM/DD h:mm a');
+      diffComment = <p className="diff-comment">{this.state.comment}&nbsp;
+                       (Edited on {revisionDateTime};&nbsp;
+                       {this.props.revision.characters} chars changed)</p>;
+    } else {
+      firstRevTime = moment(this.state.firstRevDateTime);
+      lastRevTime = moment(this.state.lastRevDateTime);
+      days = lastRevTime.diff(firstRevTime, 'days');
+      hours = lastRevTime.diff(firstRevTime, 'hours') - days * 24;
+      minutes = lastRevTime.diff(firstRevTime, 'minutes') - (days * 24 + hours) * 60;
+      if (days !== 0) {
+        timeSpan = `${days} day${days > 1 ? 's' : ''}, `;
+      }
+      if (hours !== 0) {
+        timeSpan += `${hours} hour${hours > 1 ? 's' : ''}, `;
+      }
+      if (minutes !== 0) {
+        timeSpan += `${minutes} minute${minutes > 1 ? 's' : ''}`;
+      }
+      if (!timeSpan) {
+        timeSpan = 'less than a minute';
```

Comment:
When you add an interface message, you should add it to both the en.yml and qqq.yml files. The latter is the documentation file, where you describe how the message is used within the interface (so that translators can understand the context, which is often necessary for correctly translating it, especially from a language as flexible as English).

For something like what you were trying to do, we would definitely want to use an internationalized library. (Perhaps moment.js, which we already have in the app, does this?). Rolling it by hand makes it very problematic for i18n, because it creates so-called 'lego' messages, where the whole message is built up from smaller parts. Variations in syntax from one language to another means that lego messages run into trouble when the word / phrase order needs to be switched to be correct in another language, or when gendered languages need to rely on one part of the message to get the right gender for another part.